### PR TITLE
Submitters repository

### PR DESCRIPTION
**Pull request by romanetar**
https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3207217&fileview=grid

Signed-off-by: romanetar roman_ag@hotmail.com

### CODE REVIEW

File: app/Repositories/Summit/DoctrineMemberRepository.php [hunk 1]
```diff
@@ -212,4 +688,74 @@ public function getByEmailExclusiveLock($email): ?Member
...
+                            JOIN __p.speakers __spk WITH __spk.member = e.id 
+                            JOIN __p.created_by __cb93
+                            WHERE __p.summit = :summit
+                         ) OR
+                         EXISTS (
+                            SELECT __p1.id FROM models\summit\Presentation __p1 
+                            JOIN __p1.moderator __md WITH __md.member = e.id 
+                            JOIN __p1.created_by __cb94
+                            WHERE __p1.summit = :summit
+                          )")
```
Comment (smarcet) [inline app/Repositories/Summit/DoctrineMemberRepository.php:750]: this is not needed
u need to join by creator , remove moderator
creator is submitter

File: app/Repositories/Summit/DoctrineMemberRepository.php [hunk 2]
```diff
@@ -212,4 +688,74 @@ public function getByEmailExclusiveLock($email): ?Member
...
+        return $this->getParametrizedAllIdsByPage(function () use ($summit) {
+            return $this->getEntityManager()->createQueryBuilder()
+                ->distinct(true)
+                ->select("e.id")
+                ->from($this->getBaseEntity(), "e")
+                // we need to have SIZE(e.presentations) > 0 OR SIZE(e.moderated_presentations) > 0 for a particular summit
+                ->where(" 
+                         EXISTS (
+                            SELECT __p.id FROM models\summit\Presentation __p 
+                            JOIN __p.speakers __spk WITH __spk.member = e.id
```
Comment (smarcet) [inline app/Repositories/Summit/DoctrineMemberRepository.php:741]: this is not needed
u need to join by creator , remove speakers join
creator is submitter

File: app/Repositories/Summit/DoctrineMemberRepository.php (hunk 1 above)
Comment (smarcet) [reply app/Repositories/Summit/DoctrineMemberRepository.php:750]: u only need one single jhoin

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Repositories/Summit/DoctrineMemberRepository.php (hunk 1 above)
Comment (romanetar) [reply app/Repositories/Summit/DoctrineMemberRepository.php]: removed

File: app/Repositories/Summit/DoctrineMemberRepository.php (hunk 2 above)
Comment (romanetar) [reply app/Repositories/Summit/DoctrineMemberRepository.php]: done

File: app/Repositories/Summit/DoctrineMemberRepository.php [hunk 3]
```diff
@@ -212,4 +688,74 @@ public function getByEmailExclusiveLock($email): ?Member
...
+     * @inheritDoc
+     */
+    public function getSubmittersIdsBySummit(Summit $summit, PagingInfo $paging_info, Filter $filter = null, Order $order = null)
+    {
+        return $this->getParametrizedAllIdsByPage(function () use ($summit) {
+            return $this->getEntityManager()->createQueryBuilder()
+                ->distinct(true)
+                ->select("e.id")
+                ->from($this->getBaseEntity(), "e")
+                // we need to have SIZE(e.presentations) > 0 OR SIZE(e.moderated_presentations) > 0 for a particular summit
```
[the remark this answers is not in the document]
Comment (romanetar) [reply app/Repositories/Summit/DoctrineMemberRepository.php]: done

**Comment by romanetar** [comment]
@smarcet all review comments were resolved

File: app/ModelSerializers/SubmitterMemberSerializer.php [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            $values['alternate_presentations'] = array_merge($alternate_presentation_ids, $moderated_alternate_presentation_ids);
+        }
+
+        if (in_array('rejected_presentations', $relations) && !is_null($summit)) {
+            $rejected_presentation_ids = $speaker->getRejectedPresentationIds($summit, PresentationSpeaker::RoleSpeaker, false, [], $params['filter'] ?? null);
+            $moderated_rejected_presentation_ids = $speaker->getRejectedPresentationIds($summit, PresentationSpeaker::RoleModerator, false, [], $params['filter'] ?? null);
+            $values['rejected_presentations'] = array_merge($rejected_presentation_ids, $moderated_rejected_presentation_ids);
+        }
+
+        if (!empty($expand)) {
```
Comment (smarcet) [inline app/ModelSerializers/SubmitterMemberSerializer.php:71]: lets refactor this to use the serialization by trait

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessEmail.php [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+        if(!is_null($filter))
+            $this->filter = $filter->getOriginalExp();
+        $payload  = [];
+
+        $payload['summit_name'] = $summit->getName();
+        $payload['summit_logo'] = $summit->getLogoUrl();
+        $payload['summit_schedule_url'] = $summit->getScheduleDefaultPageUrl();
+        $payload['summit_site_url'] = $summit->getLink();
+        $payload['submitter_full_name'] = $submitter->getFullName();
```
Comment (smarcet) [inline app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessEmail.php:51]: u need to provide here the 
following collections
* accepted_presentations
* alternate_presentations
* rejected_presentations

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSubmittersApiController.php [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+                ];
+            },
+            function ($filter) use ($summit) {
+                if ($filter instanceof Filter) {
+                    $filter->addFilterCondition(FilterElement::makeEqual('summit_id', $summit->getId()));
+                }
+                return $filter;
+            },
+            function () {
+                return IMemberSerializerTypes::Submitter;
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSubmittersApiController.php:231]: this serializer does not serve purposes of the CSV
please take a look at speakers CSV counter part

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedAlternateEmail.php [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+use Illuminate\Support\Facades\Log;
+use models\main\Member;
+use models\summit\Summit;
+use utils\Filter;
+
+/**
+ * Class PresentationSubmitterSelectionProcessAcceptedAlternateEmail
+ * @package App\Jobs\Emails\PresentationSubmissions\SelectionProcess
+ */
+class PresentationSubmitterSelectionProcessAcceptedAlternateEmail extends PresentationSubmitterSelectionProcessEmail
```
Comment (smarcet) [inline app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedAlternateEmail.php:24]: u forgot to seed all templates on
SummitEmailFlowTypeSeeder and SummitEmailFlowEventSeeder
also u need to create a migration to seed all flows on each summit on DB

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review commnets

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedAlternateEmail.php (hunk 7 above)
Comment (romanetar) [reply app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedAlternateEmail.php:24]: done

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSubmittersApiController.php (hunk 6 above)
Comment (romanetar) [reply app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSubmittersApiController.php]: fixed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessEmail.php (hunk 5 above)
Comment (romanetar) [reply app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessEmail.php:74]: done

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedAlternateEmail.php [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        Summit  $summit,
+        Member  $submitter,
+        ?string $test_email_recipient,
+        ?string $confirmation_token = null,
+        ?Filter $filter = null
+    )
+    {
+        parent::__construct($summit, $submitter, $test_email_recipient, $filter);
+
+        if (!empty($confirmation_token)) {
```
Comment (smarcet) [inline app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedAlternateEmail.php:54]: this is not needed there is no such feature

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedOnlyEmail.php [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        Summit  $summit,
+        Member  $submitter,
+        ?string $test_email_recipient,
+        ?string $confirmation_token = null,
+        ?Filter $filter = null
+    )
+    {
+        parent::__construct($summit, $submitter, $test_email_recipient, $filter);
+
+        if (!empty($confirmation_token)) {
```
Comment (smarcet) [inline app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedOnlyEmail.php:54]: not neeed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedRejectedEmail.php [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        Summit  $summit,
+        Member  $submitter,
+        ?string $test_email_recipient,
+        ?string $confirmation_token = null,
+        ?Filter $filter = null
+    )
+    {
+        parent::__construct($summit, $submitter, $test_email_recipient, $filter);
+
+        if (!empty($confirmation_token)) {
```
Comment (smarcet) [inline app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedRejectedEmail.php:54]: not needed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateOnlyEmail.php [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        Summit  $summit,
+        Member  $submitter,
+        ?string $test_email_recipient,
+        ?string $confirmation_token = null,
+        ?Filter $filter = null
+    )
+    {
+        parent::__construct($summit, $submitter, $test_email_recipient, $filter);
+
+        if (!empty($confirmation_token)) {
```
Comment (smarcet) [inline app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateOnlyEmail.php:54]: not needed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateOnlyEmail.php [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+     * @param string|null $test_email_recipient
+     * @param string|null $confirmation_token
+     * @param Filter|null $filter
+     */
+    public function __construct
+    (
+        Summit  $summit,
+        Member  $submitter,
+        ?string $test_email_recipient,
+        ?string $confirmation_token = null,
```
Comment (smarcet) [inline app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateOnlyEmail.php:48]: not needed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateRejectedEmail.php [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+        Member  $submitter,
+        ?string $test_email_recipient,
+        ?string $confirmation_token = null,
+        ?Filter $filter = null
+    )
+    {
+
+        parent::__construct($summit, $submitter, $test_email_recipient, $filter);
+
+        if (!empty($confirmation_token)) {
```
Comment (smarcet) [inline app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateRejectedEmail.php:55]: not needed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateRejectedEmail.php [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+     * @param string|null $test_email_recipient
+     * @param string|null $confirmation_token
+     * @param Filter|null $filter
+     */
+    public function __construct
+    (
+        Summit  $summit,
+        Member  $submitter,
+        ?string $test_email_recipient,
+        ?string $confirmation_token = null,
```
Comment (smarcet) [inline app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateRejectedEmail.php:48]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 15]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+     * @param Summit $summit
+     * @param bool $exclude_privates_tracks
+     * @param array $excluded_tracks
+     * @param Filter|null $filter
+     * @return int|mixed|string
+     */
+    public function getAcceptedPresentations
+    (
+        Summit  $summit,
+        bool    $exclude_privates_tracks = true,
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2509]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 16]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+        array   $excluded_tracks = [],
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2510]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 17]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+        ?Filter $filter = null
+    )
+    {
+        $private_tracks = [];
+
+        if ($exclude_privates_tracks) {
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2516]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 18]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+     * @param Summit $summit
+     * @param bool $include_sub_roles
+     * @param array $excluded_tracks
+     * @param Filter|null $filter
+     * @return array
+     */
+    public function getAcceptedPresentationIds
+    (
+        Summit  $summit,
+        bool    $include_sub_roles = false,
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2636]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 19]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+        array   $excluded_tracks = [],
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2637]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 20]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+     * @param Summit $summit
+     * @param bool $exclude_privates_tracks
+     * @param array $excluded_tracks
+     * @param Filter|null $filter
+     * @return bool
+     */
+    public function hasAcceptedPresentations
+    (
+        Summit  $summit,
+        bool    $exclude_privates_tracks = true,
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2659]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 21]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+        array   $excluded_tracks = [],
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2660]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 22]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+     * @param Summit $summit
+     * @param array $excluded_tracks
+     * @param false $published_ones
+     * @param Filter|null $filter
+     * @return bool
+     */
+    public function hasAlternatePresentations
+    (
+        Summit  $summit,
+        array   $excluded_tracks = [],
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2677]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 23]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+     * @param Summit $summit
+     * @param array $excluded_tracks
+     * @param bool $published_ones
+     * @param Filter|null $filter
+     * @return array
+     */
+    public function getAlternatePresentations
+    (
+        Summit $summit,
+        array $excluded_tracks = [],
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2695]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 24]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+     * @param Summit $summit
+     * @param array $excluded_tracks
+     * @param bool $published_ones
+     * @param Filter|null $filter
+     * @return array
+     */
+    public function getAlternatePresentationIds
+    (
+        Summit  $summit,
+        array   $excluded_tracks = [],
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2799]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 25]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+     * @param Summit $summit ,
+     * @param bool $include_sub_roles
+     * @param array $excluded_tracks
+     * @param Filter|null $filter
+     * @return bool
+     */
+    public function hasRejectedPresentations
+    (
+        Summit  $summit,
+        bool    $include_sub_roles = false,
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2822]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 26]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+        array   $excluded_tracks = [],
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2823]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 27]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+     * @param Summit $summit ,
+     * @param bool $exclude_privates_tracks
+     * @param array $excluded_tracks
+     * @param Filter|null $filter
+     * @return array
+     */
+    public function getRejectedPresentations
+    (
+        Summit $summit,
+        $exclude_privates_tracks = true,
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2840]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 28]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+        array $excluded_tracks = [],
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2841]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 29]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+     * @param Summit $summit
+     * @param bool $include_sub_roles
+     * @param array $excluded_tracks
+     * @param Filter|null $filter
+     * @return array
+     */
+    public function getRejectedPresentationIds
+    (
+        Summit  $summit,
+        bool    $include_sub_roles = false,
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2962]: not needed

File: app/Models/Foundation/Main/Member.php [hunk 30]
```diff
@@ -2484,4 +2496,479 @@ public function setGender(string $gender): void
...
+        array   $excluded_tracks = [],
```
Comment (smarcet) [inline app/Models/Foundation/Main/Member.php:2963]: not needed

File: database/migrations/model/Version20230227172459.php [hunk 31]
```diff
@@ -0,0 +1,42 @@
+<?php namespace Database\Migrations\Model;
```
Comment (smarcet) [inline database/migrations/model/Version20230227172459.php:1]: this migration is not correct lets do what this
https://github.com/OpenStackweb/summit-api/blob/f1d97e47f3f0efca97ad7ddf11224fdc14418cde/database/migrations/model/Version20230227172459.php
and this
https://github.com/OpenStackweb/summit-api/blob/main/database/migrations/model/Version20230125230903.php

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedOnlyEmail.php (hunk 9 above)
Comment (romanetar) [reply app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedOnlyEmail.php:54]: removed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedRejectedEmail.php (hunk 10 above)
Comment (romanetar) [reply app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAcceptedRejectedEmail.php:54]: removed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateOnlyEmail.php (hunk 11 above)
Comment (romanetar) [reply app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateOnlyEmail.php:54]: removed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateRejectedEmail.php (hunk 13 above)
Comment (romanetar) [reply app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateRejectedEmail.php:55]: removed

File: app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateRejectedEmail.php (hunk 14 above)
Comment (romanetar) [reply app/Jobs/Emails/PresentationSubmissions/SelectionProcess/PresentationSubmitterSelectionProcessAlternateRejectedEmail.php:48]: removed

File: app/Models/Foundation/Main/Member.php (hunk 15 above)
Comment (romanetar) [reply app/Models/Foundation/Main/Member.php:2509]: removed

File: app/Models/Foundation/Main/Member.php (hunk 16 above)
Comment (romanetar) [reply app/Models/Foundation/Main/Member.php:2510]: removed

File: app/Models/Foundation/Main/Member.php (hunk 17 above)
Comment (romanetar) [reply app/Models/Foundation/Main/Member.php:2516]: removed

File: app/Models/Foundation/Main/Member.php (hunk 18 above)
Comment (romanetar) [reply app/Models/Foundation/Main/Member.php:2636]: removed

File: app/Models/Foundation/Main/Member.php (hunk 19 above)
Comment (romanetar) [reply app/Models/Foundation/Main/Member.php:2637]: removed

File: app/Models/Foundation/Main/Member.php (hunk 20 above)
Comment (romanetar) [reply app/Models/Foundation/Main/Member.php:2659]: removed

File: app/Models/Foundation/Main/Member.php (hunk 21 above)
Comment (romanetar) [reply app/Models/Foundation/Main/Member.php:2660]: removed

File: app/Models/Foundation/Main/Member.php (hunk 22 above)
Comment (romanetar) [reply app/Models/Foundation/Main/Member.php:2677]: removed

File: app/Services/Model/Imp/Traits/ParametrizedSendEmails.php [hunk 32]
```diff
@@ -0,0 +1,157 @@
...
+     * @param callable $getIdsBySummit
+     * @param callable $processCurrentId
+     * @param Filter|null $filter
+     * @throws ValidationException
+     */
+    private function _sendEmails(
+        int $summit_id,
+        ISummitRepository $summit_repository,
+        array $payload,
+        ITransactionService $tx_service,
```
Comment (smarcet) [inline app/Services/Model/Imp/Traits/ParametrizedSendEmails.php:44]: why u need this u just assume that u have on this scope

File: app/Services/Model/Imp/SpeakerService.php [hunk 33]
```diff
@@ -1313,116 +1308,42 @@ public function triggerSendEmails(Summit $summit, array $payload, $filter = null
...
-
-            $email_strategy = new SpeakerActionsEmailStrategy($summit, $flow_event);
-
-            foreach ($ids as $speaker_id) {
-
+        $this->_sendEmails(
+            $summit_id,
+            $this->summit_repository,
+            $payload,
+            $this->tx_service,
```
Comment (smarcet) [inline app/Services/Model/Imp/SpeakerService.php:1315]: you dont need this param, its on this context , just use it on trait as $this->tx_service

File: app/Services/Model/Imp/SubmitterService.php [hunk 34]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function sendEmails(int $summit_id, array $payload, Filter $filter = null): void
+    {
+        $this->_sendEmails(
+            $summit_id,
+            $this->summit_repository,
```
Comment (smarcet) [inline app/Services/Model/Imp/SubmitterService.php:84]: just use it as $this->summit_repository on trait, dont pass by param

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**